src/vscode-bicep/src/commands/findOrCreateActiveBicepFile.ts

```diff
@@ -15,99 +15,115 @@ import * as fse from "fs-extra";
 import { TextDocument, Uri, window, workspace } from "vscode";
 
 type TargetFile =
-  | "rightClick"
-  | "activeEditor"
-  | "noWorkspaceActiveEditor"
+  | "rightClickOrMenu"
   | "singleInWorkspace"
   | "singleInVisibleEditors"
-  | "fromWorkspace"
+  | "quickPick"
   | "new";
 type Properties = TelemetryProperties & { targetFile: TargetFile };
 
-// Throws user-cancelled on cancel
+/**
+ * Determines which bicep file to target for a command.
+ * @summary
+ * In all cases, if a URL is passed in, it will be targeted (this handles all scenarios except invoking through a shortcut key or the command palette, e.g.
+ *   context menus will always pass in a URI).
+ * For shortcut keys and command palette, the behavior is:
+ *   1) If there's only a single bicep command available in the current workspace or visible editors, use that.
+ *   2) If there are no bicep files in the workspace and visible editors, ask the user if they want to create a new file.
+ *   3) Finally, show a pick list of available bicep files and allow them to choose (if the current editor is a bicep file, it will be at the top of the list be we will still ask).
+ * @throws User-cancelled error
```

Comment:
This would show additional error message in output channel. Can we avoid that?

With changes in this PR:
![image](https://user-images.githubusercontent.com/30270536/161662436-2c46295d-023b-41fb-b860-6a2ed8e38b66.png)

Current behavior:
![image](https://user-images.githubusercontent.com/30270536/161662498-5e6b5153-816d-44a6-9d51-dd58521438cf.png)

Repro steps:
- Invoke deploy from command palette
- Select active editor option
- Select subscription
- Hit esc and cancel

Reply:
Looks like I was testing with two different target scopes. This issue is unrelated to changes in this PR.  Sorry about the confusion.